frontend/src/modules/EditZing/Components/EmailTemplateButtons.tsx

```diff
@@ -1,21 +1,31 @@
 import { Box, Typography } from '@mui/material'
 import { RadioButtons } from '@core/Components'
-import { TemplateName } from 'EditZing/utils/emailTemplates'
 import { TemplateRadioButtonsProps } from 'EditZing/Types/ComponentProps'
 
 export const EmailTemplateButtons = ({
+  templates,
   selectedTemplate,
   setSelectedTemplate,
+  templateName,
+  setTemplateName,
 }: TemplateRadioButtonsProps) => {
   // we could include all, but designers advise to limit it to just these few for now
-  const activeTemplates = [
-    TemplateName.MATCHED,
-    TemplateName.CHECK_IN,
-    TemplateName.ADD_STUDENT,
-  ]
+  // const activeTemplates = [
+  //   TemplateName.MATCHED,
+  //   TemplateName.CHECK_IN,
+  //   TemplateName.ADD_STUDENT,
+  // ]
+
+  const templateNames = templates.map((template) => template.name)
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setSelectedTemplate(event.target.value as TemplateName)
+    const choice = event.target.value
+    templates.forEach((template) => {
+      if (template.name === choice) {
+        setSelectedTemplate(template)
+      }
+    })
+    setTemplateName(event.target.value)
```

Comment:
The reason why keying things on data like names instead of ids is generally not preferable is because there's no guarantee the template names are unique, unlike ids. But I can see how that would be difficult in this case because the RadioButtons component is really old and doesn't distinguish between values and labels.

Reply:
Yeah, I can see how this is actually a problem but I don't see how I could change the selected template any other way unless I rebuild the radio buttons/use the default radio buttons (not in our core lib) that can take in the id (to handle change) and name (to display). Is there any way to fix it as it is now to check id against the selected radio button?

Reply:
I think modifying `RadioButtons.tsx` would not be too hard, as the relevant existing code is just the following. You could just add another prop called `labels` and associate the values to labels by their index.

```jsx
<RadioButton
  value={value}
  label={value}
/>
```